packages/product-components/src/components/SelectAssetModal/SelectAssetModal.tsx

```diff
@@ -34,110 +28,43 @@ export interface SelectAssetOptionGroupProps {
-    name: Network['name'];
-    symbol: Network['symbol'];
-    coingeckoId: Network['coingeckoId'];
-    coingeckoNativeId?: Network['coingeckoNativeId'];
-}
-
-export type SelectAssetSearchCategoryType = {
-    coingeckoId: string;
-    coingeckoNativeId?: string;
-} | null;
-
-export interface SelectAssetModalProps {
-    options: SelectAssetOptionProps[];
-    networkCategories: SelectAssetNetworkProps[];
-    onSelectAssetModal: (selectedAsset: string) => void;
-    onFavoriteClick?: (isFavorite: boolean) => void;
-    onClose: () => void;
-}
-
-interface AssetProps
-    extends Pick<SelectAssetOptionCurrencyProps, 'coingeckoId' | 'contractAddress'> {
-    cryptoId: string;
-    symbol: string;
-    name: string;
+export interface AssetProps
+    extends Pick<
+        SelectAssetOptionCurrencyProps,
+        'symbol' | 'badge' | 'networkSymbol' | 'cryptoName' | 'coingeckoId' | 'contractAddress'
+    > {
     height: number;
-    isFavorite: boolean;
-    badge?: string;
+    shouldTryToFetch?: boolean;
 }
 
+export const ITEM_HEIGHT = 60;
+const LIST_MIN_HEIGHT = 200;
 const HEADER_HEIGHT = 267;
-const ITEM_HEIGHT = 60;
 const LIST_HEIGHT = `calc(80vh - ${HEADER_HEIGHT}px)`;
-const LIST_MIN_HEIGHT = ITEM_HEIGHT * 3;
-
-const getData = (options: SelectAssetOptionProps[]): AssetProps[] =>
-    options
-        .filter(item => item.type === 'currency')
-        .map(item => ({
-            cryptoId: item.value,
-            symbol: item.label,
-            name: item.cryptoName ?? item.label,
-            badge: item.networkName,
-            coingeckoId: item.coingeckoId,
-            contractAddress: item.contractAddress,
-            isFavorite: false,
-            height: ITEM_HEIGHT,
-        }));
-
-const getNetworkCount = (options: SelectAssetOptionProps[]) => {
-    const networkNetworkGroups = options
-        .filter(item => item.type === 'group' && item.networkName)
-        .map(networkGroup => ({
-            ...networkGroup,
-            coingeckoNativeId: networkGroup.coingeckoId
-                ? getNetworkByCoingeckoId(networkGroup.coingeckoId)?.coingeckoNativeId
-                : undefined,
-        }));
-    const networkCurrencies = options.filter(
-        item =>
-            item.type === 'currency' &&
-            !item.contractAddress &&
-            !networkNetworkGroups.find(
-                group =>
-                    group.coingeckoId === item.coingeckoId ||
-                    group.coingeckoNativeId === item.coingeckoId,
-            ),
-    );
 
-    return networkNetworkGroups.length + networkCurrencies.length;
-};
+export interface SelectAssetModalProps {
+    options: AssetProps[];
+    searchPlaceholderText?: string;
+    onSelectAsset: (selectedAsset: Omit<AssetProps, 'height'>) => Promise<void> | void;
+    onClose: () => void;
+    searchInput?: ReactNode;
+    filterTabs?: ReactNode;
+    noItemsAvailablePlaceholder: { heading: string | ReactNode; body?: string | ReactNode };
```

Comment:
`string` is assignable to `ReactNode`
```suggestion
    noItemsAvailablePlaceholder: { heading: ReactNode; body?: ReactNode };
```